addons/web_tour/static/src/tour_service/tour_step_automatic.js

```diff
@@ -15,26 +15,32 @@ export class TourStepAutomatic extends TourStep {
         this.tourConfig = tourState.getCurrentConfig();
     }
 
-    async checkForUndeterminisms() {
-        const delay = this.tourConfig.delayToCheckUndeterminisms;
-        if (delay > 0 && this.element) {
-            const snapshot = this.element.cloneNode(true);
-            return new Promise((resolve, reject) => {
-                browser.setTimeout(() => {
-                    if (this.element.isEqualNode(snapshot)) {
-                        resolve();
-                    } else {
-                        reject(
-                            new Error(
-                                [
-                                    ...this.describeWhyIFailed,
-                                    `UNDETERMINISM: two differents elements have been found in ${delay}ms for trigger ${this.trigger}`,
-                                ].join("\n")
-                            )
-                        );
-                    }
-                }, delay);
-            });
+    async checkForUndeterminisms(initialElement, delay) {
+        if (delay > 0 && initialElement) {
```

Comment:
When functions are composed of single conditional blocks, it is generally cleaner to invert the condition and add a return statement to spare an indentation level and improve clarity
```suggestion
        if (delay <= 0 || !initialElement) {
            return;
        }
```